Purge button

jjeffryes opened the pull request:
This adds a purge button to settings/advanced that will remove all shared files from your server. This addresses concerns of users in areas that prohibit distribution of some content, if those users inadvertently view prohibited content they can clear their cache so it's not shared with other users over IPFS.

rmisio commented on js/views/modals/Settings/Advanced.js:
```diff
@@ -47,6 +48,35 @@ export default class extends baseVw {
+      });
+  }
+
+
```
Don't forget the super.render().

At some point, I think I'll try and write a test that would check all views and ensure that if they are using getCachedEl, that they are calling super.render(). It's really easy to forget.

rmisio replied:
It might not be a bad idea to pass `isPurging: this.purge && this.purge.state() === 'pending'` into the template and then use that in the template to decide whether the `processing` class should be applied to the button.

Otherwise, if you re-render during a purge (e.g. user clicks Save), then the spinner state is lost and the user might just kick off another purge.

rmisio commented on js/languages/en-US.json:
```diff
@@ -358,9 +356,17 @@
+        "purgeBtn": "Purge Files",
+        "purgeToolTip": "OpenBazaar shares content you have viewed with other users. If you have viewed text or images you don't want to share this button will delete them. This will not affect any files that belong to you.",
+        "purgeError": "There was an error, your cached files may not have been deleted.",
+        "purgeComplete": "Your cached files have been deleted."
```
You may want to check with @cpacia to know if 'ob/purgecache' returns after the purge is complete or if it just kicks off a purge, returns immediately, and the purge finished at some time. That's how the publish api works. If the purge API works the same way, I think we'd want to tweak the messaging since `Your cached files have been deleted.` wouldn't be completely accurate.

jjeffryes commented:
@rmisio this is updated. I also updated the save button with a saving state, so it will maintain it's state the same way the purging button will, and updated the save button to use getCachedEl.

Chris confirmed the purge api returns after it has completed. It usually only takes 40-50ms, but in theory could be longer.

rmisio commented on js/views/modals/Settings/Advanced.js:
```diff
@@ -95,40 +135,36 @@ export default class extends baseVw {
-  }
-
   render() {
     loadTemplate('modals/settings/advanced.html', (t) => {
```
Don't forget the super.render() if using getCachedEl.
